scene/main/viewport.cpp

```diff
@@ -2789,7 +2789,9 @@ bool Viewport::_sub_windows_forward_input(const Ref<InputEvent> &p_event) {
 
 	gui.subwindow_focused->_window_input(ev);
 
-	return true;
+	if(!gui.subwindow_focused)
+		return true;
+	return gui.subwindow_focused->is_input_handled() || gui.subwindow_focused->is_handling_input_locally();
```

Comment:
This is a minor thing, but we generally don't do bracketless if statements. Additionally, the code flow might be a bit easier to grasp with the following modification:

```suggestion
	if (gui.subwindow_focused) {
            return gui.subwindow_focused->is_input_handled() || gui.subwindow_focused->is_handling_input_locally();
        }
        
        return true;
```

The code flow part is mostly personal though. Otherwise, nice work and it seems to work well on my end.